lib/CUDAKernels/src/CUDAKernels.jl

```diff
@@ -53,6 +54,49 @@ function next_stream()
         return stream
     end
 end
+=#
+const FREE_STREAMS_D = Dict{CUDA.CuContext,Array{CUDA.CuStream,1}}()
+const STREAMS_D      = Dict{CUDA.CuContext,Array{CUDA.CuStream,1}}()
+function next_stream()
+    ctx = CUDA.current_context()
+    lock(STREAM_GC_LOCK) do
+        # see if there is a compatible free stream
```

Comment:
```
FREE_STREAMS_CT  = get!(FREE_STREAMS_D, ctx) do
    CUDA.CuStream[]
end
if !isempty(FREE_STREAMS_CT)
  return pop!(FREE_STREAMS_CT)
end
```

Reply:
Yes, `get!` executes the function when there is no key present, and uses it return value to fill it in.

Reply:
I guess it does 
```
FREE_STREAMS_D=Dict()
ctx=3
FREE_STREAMS_D

>> Dict{Any, Any}()

FREE_STREAMS_CT  = get!(FREE_STREAMS_D, ctx) do
    7
end
FREE_STREAMS_D

>> Dict{Any, Any} with 1 entry:
>>  3 => 7

```